### PR TITLE
boards/nucleo-f334r8: improvements to documentation

### PR DESCRIPTION
**Pull request by krzysztof-cabaj**
### Contribution description

Improvements to STM Nucleo f336r8 documentation:

- Fix wrong parameter in command used for flashing
- Add information concerning accessing RIOT shell (see Issue #17453) 

### Testing procedure

```
make doc
xdg-open doc/doxygen/html/group__boards__nucleo-f334r8.html
```

### Issues/PRs references

Issue #17453

### CODE REVIEW

File: boards/nucleo-f334r8/doc.txt [hunk 1]
```diff
@@ -77,11 +77,26 @@ could be performed manually; however, the cpy2remed (copy to removable
...
 
+## Accessing RIOT shell
+
+Default RIOT shell access utilize VCP (Virtual COM Port) via USB interface,
+provided by integrated ST-LINK programmer. ST-LINK is connected to the
+microcontroller USART2.
+
+The default baud rate is 115200.
+
+If a physical connection to USART2 is needed, connect UART interface to pins
```
Comment (chrysn) [inline boards/nucleo-f334r8/doc.txt:93]: Won't a physical user of these pins drive PA3 against the programmer?

Comment (krzysztof-cabaj) [reply boards/nucleo-f334r8/doc.txt:93]: I don't know if I understand you correctly. But yes, these pins are connected to the ST-Link programmer. But if virtual com is not used they can be directly connected to other UART.

Comment (chrysn) [reply boards/nucleo-f334r8/doc.txt:93]: [Board manual](https://www.st.com/resource/en/user_manual/dm00105823-stm32-nucleo-64-boards-mb1136-stmicroelectronics.pdf) indicates there are solder bridges that'd need to be switched over to actually use PA[23] through the external D[10] pins -- so I'd suggest to just leave it at USART2 being connected to the debugger, or be explicit about board modifications being required to use the D[10] pins.

(Alternatively you could point out that UART_DEV(1) is available at pins A9/10 and UART_DEV(2) is available at B10/11 in some configurations, and that for non-USB stdio the `STDIO_UART_DEV` could be changed to them, although personally I think that documenting the default stdio is good enough and that any alternative stdio configs are up to the user).

Comment (krzysztof-cabaj) [reply boards/nucleo-f334r8/doc.txt:93]: Indeed, these solder bridges - connect PA[23] to D[10].  But in my humble opinion, accordingly to figure 17 PA[23] are connected to the ST-Link and pins 35 and 37 at CN10. Virtual com is great for programming, but sometimes direct connection to stdio is beneficial - I often use this for some initial configuration when board is connected to external power source, and after configuration I move board. 

This is my rationale for this information, and I would like leave it. And I agree that more complicated stdio access should be described in the other place.

Comment (krzysztof-cabaj) [reply boards/nucleo-f334r8/doc.txt]: I take f334 from lab and experiment with it. You are right - without changes of solder bridges there are not connection to CN10 pins 35 and 37. I was confused by schematic in the nucleo board box, where these pins are labeled PA2 and PA3. 
I remove wrong information concerning physical access.

**Comment by benpicco** [comment]
Please squash

**Pull request merged**